kombu/transport/redis.py

```diff
@@ -562,6 +569,23 @@ def __init__(self, *args, **kwargs):
             # by default.
             self.keyprefix_fanout = ''
 
+        # Prepend the global key prefix
+        self.unacked_key = self._queue_with_prefix(self.unacked_key)
+        self.unacked_index_key = self._queue_with_prefix(
+            self.unacked_index_key
+        )
+        self.unacked_mutex_key = self._queue_with_prefix(
+            self.unacked_mutex_key
+        )
```

Comment:
@gabor-boros 

The final step isn't working for me as described?

> 7. Run snippet "example 2" (defined below) and validate the message object is printed 👍 
> 8. List all keys (`KEYS *`) and validate (only) `_prefixed_unacked`, `_prefixed_unacked_index`, and `_prefixed__kombu.binding.default` are listed

I'm seeing this instead:
```
127.0.0.1:6379> keys *
1) "a_queue_name"
2) "_prefixed__kombu.binding.default"
```

Any idea why?

Reply:
@pomegranited I have no idea, as it is running fine for me.

Reply:
Ok, maybe my virtualenv setup was different somehow?

Here's what I trued -- do you see any differences with yours that might be relevant?

```bash
$ python3.8 -m venv ~/.virtualenvs/redis-celery
$ source ~/.virtualenvs/redis-celery/bin/activate
(redis-celery) $ python --version
Python 3.8.3

# Started by using the versions of `redis` and `celery` taken from
# https://github.com/edx/edx-platform/blob/master/requirements/edx/base.txt 
(redis-celery) $ cat requirements_edx.txt
django-celery-results<2.1  # brings in supported celery
redis==3.5.3
-e git+https://github.com/open-craft/kombu.git@gabor/add-global-keyprefix#egg=kombu==gabor/add-global-keyprefix

# But the same issue happens when I'm using the latest versions of `redis` and `celery`:
(redis-celery) $ cat requirements.txt
(redis-celery) helga:global_keys$ cat requirements.txt 
celery
redis
-e git+https://github.com/open-craft/kombu.git@gabor/add-global-keyprefix#egg=kombu==gabor/add-global-keyprefix

(redis-celery) $ pip install -r requirements.txt
(redis-celery) $ python ex2.py
<Message object at 0x7faeeac2b550 with details {'state': 'RECEIVED', 'content_type': 'text/plain', 'delivery_tag': 'f0a7daa4-7813-4e3d-aeb7-1047b47b6c94', 'body_length': 9, 'properties': {}, 'delivery_info': {'exchange': 'default', 'routing_key': ''}}>
Restoring 1 unacknowledged message(s)

# Meanwhile, over at the redis-cli:
127.0.0.1:6379> keys *
1) "_prefixed__kombu.binding.default"
2) "a_queue_name"
```


```

Reply:
I did find the issue. Will push a commit later today fixing the problem.

Reply:
@gabor-boros You're brilliant, that fixed it! Will do another code review pass and approve this. Thank you!

Reply:
@gabor-boros Thanks for updating this PR, but I'm unable to get the example scripts you wrote to print out any messages.

I used the [same setup as before](https://github.com/celery/kombu/pull/1349#discussion_r655881808), but when I run the scripts, I get no messages dequeued, and no output printed from either ex1.py or ex2.py:

```
(redis-celery) $ python ex1.py
(no output)
```

The keys are showing properly prefixed in redis though:
```
127.0.0.1:6379> keys *
1) "_prefixed__kombu.binding.default"
2) "_prefixed_a_queue_name"
```

Are you seeing the same thing?

Reply:
@pomegranited I see the same, though it is strange. I tested the changes only with the latest script provided by @matusvalo.

@matusvalo do you have any clue why the `Consumer` is not calling the `on_message` callback at all in the scripts that are provided as part of the PR description?

Reply:
Or @matusvalo do you have any idea how to properly test the scenarios other than the script you provided?